### DISCUSSION TITLE
Show playhead cursor on play, hide it on stop.

### DISCUSSION THREAD

**Discussion by sojastar** [katspaugh/wavesurfer.js#3581] [Q&A]
Hello,

First, a big thank you for this wonderful library! 🙏 🙇‍♂️ 

I would like to hide the playhead cursor when the sound is playing, and hide it when the sound is paused or stoped. I have tried to change the width or color of the cursor with `setOptions` like this (but without much success):
```javascript
function play(wavesurfer) {
  wavesurfer.play();
  wavesurfer.setOptions = {
    container: wavesurfer,
    cursorWidth: 8
  };
}

function stop(wavesurfer) {
  wavesurfer.stop();
  wavesurfer.setOptions = {
    container: wavesurfer,
    cursorWidth: 0
  };
}
```
Any idea on what I am doing wrong?

Thanks!

**Comment by KaleSurfer** [comment]
I'm not an expert by any means, but a few things:
1) play() returns a promise (is async)
2) it's setOptions({ ... my options ... })
3) some code - I changed the values to make it more obvious:

```
async function play(wavesurfer) {
   await wavesurfer.play();
   wavesurfer.setOptions({
      cursorColor: "orange",
      cursorWidth: 16
   });
}

function stop(wavesurfer) {
  wavesurfer.stop();
  wavesurfer.setOptions({
    cursorColor: "blue",
    cursorWidth: 7
  });
}
```

**Comment by katspaugh** [comment]
This works for me:

```js
wavesurfer.on('play', () => {
  wavesurfer.setOptions({ cursorWidth: 8 })
})

wavesurfer.on('pause', () => {
  wavesurfer.setOptions({ cursorWidth: 0 })
})
```
